packages/wasm-dpp/src/document/document_facade.rs

```diff
@@ -95,10 +97,12 @@ impl DocumentFacadeWasm {
     pub fn create_state_transition(
         &self,
         documents: &JsValue,
-        nonce_counter_value: &js_sys::Object, //IdentityID/ContractID -> nonce
+        nonce_counter_value: &js_sys::Object, //IdentityID/ContractID -> nonce,
+        recipient: Option<IdentifierWrapper>,
+        price: Option<Credits>,
```

Comment:
This looks strange here, to be honest. For which exact document it applies? I think these params must be a part of document information

Reply:
For NFT operations, like transfer, updatePrice or purchase. Neither Document or ExtendedDocument have such fields

Reply:
Then we need to accept another structure that will contain document and params

Reply:
I just wondering why do we need document for transfer? We need just ID, isn't ?

Reply:
> Then we need to accept another structure that will contain document and params

Should I revert back to this approach?
![image](https://github.com/user-attachments/assets/e99ff400-3bf5-46d9-aebc-3dc35bfa920b)

> I just wondering why do we need document for transfer? We need just ID, isn't ?

Yes, it is possible, however there are no such methods in the rs-dpp yet (only from_document)